src/bin/simple_dql.rs

```diff
@@ -0,0 +1,159 @@
+        #[clap(long)]
+        make: String,
+
+        /// The year of the car
+        #[clap(long)]
+        year: String,
+
+        /// The color of the car
+        #[clap(long)]
+        color: String,
+    },
+    /// Query existing "car" documents based on their color property
+    QueryCars {
+        /// Query all cars with this color
+        #[clap(long)]
+        color: String,
+    },
+}
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    dotenv::dotenv().ok();
+    let cli = Cli::parse();
+
+    // Initialize Ditto SDK client
+    let args = &cli.args;
+    let app_id = AppId::from_str(&args.app_id)?;
+    let ditto = Ditto::builder()
+        .with_root(Arc::new(PersistentRoot::from_current_exe()?))
+        .with_minimum_log_level(LogLevel::Debug)
+        .with_identity(move |ditto_root| {
+            let shared_token = args.playground_token.clone();
+            let enable_cloud_sync = true;
+            let custom_auth_url = None;
+            OnlinePlayground::new(
+                ditto_root,
+                app_id,
+                shared_token,
+                enable_cloud_sync,
+                custom_auth_url,
+            )
+        })?
+        .build()?;
+
+    // Begin sync, then open the Ditto Store so we can insert or query documents
+    ditto.start_sync()?;
+    let store = ditto.store();
+
+    match cli.cmd {
+        Cmd::CreateCar { make, year, color } => {
+            let result_set = create_car(store, &make, &year, &color).await?;
+            let mutations = result_set.mutated_document_ids();
+            let s = if mutations.len() == 1 { "" } else { "s" };
+            println!("Mutated {} document{s}", mutations.len());
+        }
+        Cmd::QueryCars { color } => {
+            let result_set = query_cars(store, &color).await?;
+            let count = result_set.item_count();
+            println!("Found {count} cars with color={}", color);
+
+            // Print the contents of each queried document
+            for item in result_set.iter() {
+                let json = item.deserialize_value::<serde_json::Value>()?;
+                println!("Car with color={}: {json}", color);
+            }
+        }
+    }
+
+    Ok(())
+}
+
+async fn create_car(store: &Store, make: &str, year: &str, color: &str) -> Result<QueryResult> {
+    let result_set = store
+        .execute(
+            "INSERT INTO cars DOCUMENTS (:newCar)",
+            Some(
+                serde_json::json!({
+                    "newCar": {
+                        "make": make,
+                        "year": year,
+                        "color": color
+                    }
+                })
+                .into(),
+            ),
+        )
+        .await?;
+
+    Ok(result_set)
+}
+
+async fn query_cars(store: &Store, color: &str) -> Result<QueryResult> {
+    let query_args = QueryArguments::from(serde_json::json!({
+        "color": color,
+    }));
```

Comment:
Tiny æsthetical preference for moving the `.into()` to the callsite itself:

```suggestion
    let query_args = serde_json::json!({
        "color": color,
    });
```